### ISSUE TITLE
Static Content Deploy in developer mode without force flag is broken in 2.3-develop

### ISSUE THREAD

**Issue by hostep** [magento/magento2#24678]
### Preconditions (*)
1. Magento 2.3-develop branch, I tested with commit 58600419a766eabcf180781aa80b6bfd12daedb3
2. PHP 7.2.20

### Steps to reproduce (*)
1. Setup 2.3-develop installation
2. Make sure it's in developer mode
3. Run `bin/magento setup:static-content:deploy`

### Expected result (*)
1. Getting useful error:
```
$ bin/magento setup:static-content:deploy

In DeployStaticContentCommand.php line 108:

  NOTE: Manual static content deployment is not required in "default" and "developer" modes.
  In "default" and "developer" modes static contents are being deployed automatically on demand.
  If you still want to deploy in these modes, use -f option: 'bin/magento setup:static-content:deploy -f'


setup:static-content:deploy [-f|--force] [-s|--strategy [STRATEGY]] [-a|--area [AREA]] [--exclude-area [EXCLUDE-AREA]] [-t|--theme [THEME]] [--exclude-theme [EXCLUDE-THEME]] [-l|--language [LANGUAGE]] [--exclude-language [EXCLUDE-LANGUAGE]] [-j|--jobs [JOBS]] [--max-execution-time [MAX-EXECUTION-TIME]] [--symlink-locale] [--content-version CONTENT-VERSION] [--refresh-content-version-only] [--no-javascript] [--no-css] [--no-less] [--no-images] [--no-fonts] [--no-html] [--no-misc] [--no-html-minify] [--] [<languages>...]
```

### Actual result (*)
1. Getting very obscure error:
```
#0 vendor/symfony/console/Command/Command.php(255): Magento\Setup\Console\Command\DeployStaticContentCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 vendor/symfony/console/Application.php(893): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 vendor/symfony/console/Application.php(262): Symfony\Component\Console\Application->doRunCommand(Object(Magento\Setup\Console\Command\DeployStaticContentCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 lib/internal/Magento/Framework/Console/Cli.php(107): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 vendor/symfony/console/Application.php(145): Magento\Framework\Console\Cli->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 bin/magento(23): Symfony\Component\Console\Application->run()
#6 {main}
```

### Further findings
After git bisecting, it revealed commit https://github.com/magento/magento2/commit/9ffe7eb2d3e6fcbf18eb39b6b847e99b0550a028 to have caused the problem, this is part of PR https://github.com/magento/magento2/pull/24044

@PascalBrouwers: maybe you could have a look at what's going on here? Thanks! 🙂
Is not super important but might be confusing for newcomers in the Magento world.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20327` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hostep** [comment]
@Alex-Sy: fix is already in progress: https://github.com/magento/magento2/pull/24734

**Comment by magento-engcom-team** [comment]
Hi @hostep. Thank you for your report. 
The issue has been fixed in magento/magento2#24734 by @p-bystritsky in 2.3-develop branch
Related commit(s):
 - [dfd481d22616708a7e55d7b8757581e90ba4f41c](https://github.com/magento/magento2/commit/dfd481d22616708a7e55d7b8757581e90ba4f41c)

The fix will be available with the upcoming 2.3.4 release.
